examples/api/lib/widgets/app_model/app_model.1.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for AppModel
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+// A single lazily constructed object that's shared with the entire
+// application via `SharedObject.of(context)`. The value of the object
+// can be changed with `SharedObject.reset(context)`. Resetting the value
+// will cause all of the widgets that depend on it to be rebuilt.
+class SharedObject {
+  SharedObject._();
+
+  static final Object _sharedObjectKey = Object();
+
+  @override
+  String toString() => describeIdentity(this);
+
+  static void reset(BuildContext context) {
+    // Calling AppModel.setValue() causes dependent widgets to be rebuilt.
+    AppModel.setValue<Object, SharedObject>(context, _sharedObjectKey, SharedObject._());
+  }
+
+  static SharedObject of(BuildContext context) {
+    SharedObject? value = AppModel.getValue<Object, SharedObject>(context, _sharedObjectKey);
```

Comment:
I think a "putIfAbsent" api makes more sense

Reply:
The primary use case for the API is looking up values so I think a method name based on `get` is OK.